services/directions.js

```diff
@@ -0,0 +1,151 @@
+'use strict';
+
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+
+/**
+ * Directions API service.
+ */
+var Directions = {};
+
+/**
+ * Get directions.
+ *
+ * See the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#directions).
+ *
+ * @param {Object} config
+ * @param {'driving-traffic'|'driving'|'walking'|'cycling'} [config.profile=driving]
+ * @param {Array.<Object>} config.way_points - An ordered array of objects with `latitude`
+ * and `longitude` properties that represent points to visit and optional `approach`, `bearing`, `radius` & `waypoint_name`,
+ * see the [Mapbox Directions API](https://www.mapbox.com/api-documentation/#retrieve-directions) for details.
+ * There can be between 2 and 25 coordinates.
+ * @param {string} [config.alternatives=false]
+ * @param {Array.<string>} [config.annotations]
+ * @param {boolean} [config.banner_instructions=false]
+ * @param {boolean} [config.continue_straight]
+ * @param {string} [config.exclude]
+ * @param {string} [config.geometries=polyline]
+ * @param {string} [config.language=en]
+ * @param {string} [config.overview=simplified]
+ * @param {boolean} [config.roundabout_exits=false]
+ * @param {boolean} [config.steps=false]
+ * @param {boolean} [config.voice_instructions=false]
+ * @param {string} [config.voice_units=imperial]
+ * @return {MapiRequest}
+ */
+Directions.getDirections = function(config) {
+  config.profile = config.profile || 'driving';
+
+  v.validate(
+    {
+      profile: v.string,
+      way_points: v.arrayOfObjects.required,
+      alternatives: v.boolean,
+      annotations: v.arrayOfStrings,
+      banner_instructions: v.boolean,
+      continue_straight: v.boolean,
+      exclude: v.string,
+      geometries: v.string,
+      language: v.string,
+      overview: v.string,
+      roundabout_exits: v.boolean,
+      steps: v.boolean,
+      voice_instructions: v.boolean,
+      voice_units: v.string
+    },
+    config
+  );
+
+  config.way_points.forEach(function(wayPoint) {
+    v.validate(
+      {
+        latitude: v.number.required,
+        longitude: v.number.required,
+        approach: v.string,
+        bearing: v.arrayOf(v.number),
+        radius: v.oneOf([v.number, v.string]),
+        waypoint_name: v.string
+      },
+      wayPoint
+    );
+  });
+
+  var wayPoints = {
+    coordinates: [],
+    approach: [],
+    bearing: [],
+    radius: [],
+    waypoint_name: []
+  };
+
+  config.way_points.forEach(function(wayPoint) {
+    wayPoints.coordinates.push(wayPoint.longitude + ',' + wayPoint.latitude);
+
+    // join props which come in pairs
+    ['bearing'].forEach(function(prop) {
+      if (wayPoint.hasOwnProperty(prop) && wayPoint[prop] != null) {
+        wayPoint[prop] = wayPoint[prop].join(',');
```

Comment:
After 843fc2693108a427f3cfb5b888ff51aa2c59de87, we'll want this to be an array instead of a string, so the comma isn't encoded.